docs/history-framework.md

```diff
@@ -193,7 +193,113 @@ sequenceDiagram
   deactivate TreeManager
 ```
 
-## TODO:
+## Serialization
+We store the history of changes to a document in Firestore. Each history entry is stored in a separate Firestore document. These history entry documents are stored in the same way that comments are stored. There is a parent Firestore document that has meta data about the actual CLUE document and then under this parent Firestore document is a collection for the comments and a collection for the history entries.
+
+The history entries are sent up by the TreeManager which is an MST model. In other places we interact with Firestore through react components so we can use hooks. Since we are in the model we just work directly with Firestore.
+
+The history events are downloaded only when needed for replaying the history. This is done by a mostly invisible `<LoadDocumentHistory>` component. It currently puts up an ugly message on the screen to let the user know something is happening. This component uses hooks to load in the history and then update the document with this history. It is kind of strange for a React component to be managing the MST model. But this temporary history MST model is being created by the component only as it is needed, it isn't something shared by multiple components.
+
+The history loading currently doesn't do any batching/paging during the load, so if the history gets large enough this might cause problems. There is a FIXME in the code for this. The component is currently monitoring Firestore collection, so if new history events are added they will be shown immediately without having to close and open the history UI. It isn't totally clear how these incremental history entries are handled. I think the `LoadDocumentHistory` component is re-rendered when new events show up and then the history in the TreeManager gets completely replaced with the updated history. In other words the entries aren't really loaded incrementally. We'll probably need to improve this to handle documents with large histories.
+
+To create the parent document which contains the history entires in Firestore the TreeManager is using out Firebase function `validateCommentableDocument`. This can create either a document associated with a networked teacher or a generic user document. When we tackle handling permissions this document will be required to grant students and teachers access to the history entries of this document.
```

Comment:
```suggestion
To create the parent document which contains the history entries in Firestore the TreeManager is using our Firebase function `validateCommentableDocument`. This can create either a document associated with a networked teacher or a generic user document. When we tackle handling permissions it will be necessary to grant students and teachers access to the history entries of this document.
```